File: backend/nodejs/src/routers/health.ts
import { Router } from 'express';

import type { AppConfig } from '../core/config.js';

export function createHealthRouter(config: AppConfig) {
  const router = Router();

  router.get('/health', (_request, response) => {
    response.json({
      status: 'ok',
      provider: config.llmProvider,
      version: config.appVersion,
    });
  });

  return router;
}

File: backend/python/app/routers/health.py
from fastapi import APIRouter, Depends

from app.core.config import APP_VERSION, Settings, get_settings

router = APIRouter()


@router.get("/api/health")
async def health(settings: Settings = Depends(get_settings)) -> dict[str, str]:
    return {
        "status": "ok",
        "provider": settings.llm_provider,
        "version": APP_VERSION,
    }
